include/musica/micm.hpp

```diff
@@ -162,20 +207,66 @@ namespace musica
     template<class T>
     T GetSpeciesProperty(const std::string &species_name, const std::string &property_name, Error *error);
 
-    static constexpr std::size_t NUM_GRID_CELLS = 1;
+   public:
+    MICMSolver solver_type_;
 
-   private:
-    using DenseMatrixPolicy = micm::Matrix<double>;
-    using SparseMatrixPolicy = micm::SparseMatrix<double, micm::SparseMatrixStandardOrdering>;
-    using SolverPolicy = typename micm::RosenbrockSolverParameters::
-        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixPolicy, micm::LuDecomposition>>;
-    using Rosenbrock = micm::Solver<SolverPolicy, micm::State<DenseMatrixPolicy, SparseMatrixPolicy>>;
+    /// @brief Vector-ordered Rosenbrock solver type
+    using DenseMatrixVector = micm::VectorMatrix<double, MICM_VECTOR_MATRIX_SIZE>;
+    using SparseMatrixVector = micm::SparseMatrix<double, micm::SparseMatrixVectorOrdering<MICM_VECTOR_MATRIX_SIZE>>;
+    using RosenbrockVectorType = typename micm::RosenbrockSolverParameters::
+        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixVector, micm::LuDecomposition>>;
+    using Rosenbrock = micm::Solver<RosenbrockVectorType, micm::State<DenseMatrixVector, SparseMatrixVector>>;
+    std::unique_ptr<Rosenbrock> rosenbrock_;
 
-    std::unique_ptr<Rosenbrock> solver_;
+    /// @brief Standard-ordered Rosenbrock solver type
+    using DenseMatrixStandard = micm::Matrix<double>;
+    using SparseMatrixStandard = micm::SparseMatrix<double, micm::SparseMatrixStandardOrdering>;
+    using RosenbrockStandardType = typename micm::RosenbrockSolverParameters::
+        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixStandard, micm::LuDecomposition>>;
+    using RosenbrockStandard = micm::Solver<RosenbrockStandardType, micm::State<DenseMatrixStandard, SparseMatrixStandard>>;
+    std::unique_ptr<RosenbrockStandard> rosenbrock_standard_;
```

Comment:
Are we allowed to have a pointer to a `micm::Solver` without specifying the template types?

Reply:
or maybe we use `std::variant`?

Reply:
Do you mean something like `std::unique_ptr<micm::Solver> solver`? If it is the case, I tried it but it didn't work, (`template argument is invalid`). Unless`MICM` is a template class which is not a viable option, I think all the types must be specified.

I tried implementing `std::variant` and realized that you still need to do type check to get the data that variant holds, and the syntax can be quite tricky. For example, `MICM::Solve( lots of params ... )`
```c++
// Current
micm::State state = std::get<Rosenbrock>(solver)->GetState();

// std::variant
if(std::get_if<Rosenbrock>(&solver))
  micm::State state = std::get<Rosenbrock>(&solver)->GetState();
if(std::get_if<RosenbrockStandardOrder>(&solver))
  micm::State state = std::get<RosenbrockStandardOrder>(&solver)->GetState();
```

What do you think? The fact that C doesn't support any form of overloading, we quite have limitations here.

Reply:
Hmm. Good point. What you have seems fine anyways